packages/foam-vscode/src/features/tag-completion.ts

```diff
@@ -1,10 +1,16 @@
+import { EOL } from 'os';
 import * as vscode from 'vscode';
+import matter from 'gray-matter';
 import { Foam } from '../core/model/foam';
 import { FoamTags } from '../core/model/tags';
 import { HASHTAG_REGEX } from '../core/utils/hashtags';
 import { FoamFeature } from '../types';
 import { mdDocSelector } from '../utils';
 
+const MAX_FRONT_MATTER_LINES = 50;
+const FRONT_MATTER_TAG_REGEX =
+  /^\s*tags:\s+.*?\s{0,1}?#?([0-9]*[\p{L}\p{Emoji_Presentation}/_-][\p{L}\p{Emoji_Presentation}\p{N}/_-]*)$/gmu;
```

Comment:
will this work for all the tags or just the first one? also there is the question of frontmatter "flexibility", e.g. the following is valid frontmatter that creates 5 tags:

```
---
tags: aaa aai, aab
  aae
  aao
---
```

This is also valid YAML

```
---
tags: 
 - aaa aai, aab
  aae
  aao
 - aau
---
```

I would be ok with only supporting "same line" tags (seems like the majority of the cases imo), but we need to support the space and comma separators

Reply:
checked in https://github.com/foambubble/foam/pull/1191/files#diff-2f15bd61b55d617a33f7675aa163c5ed8e168efd6b92630c0f5f16f37f7c470bR247